internal/webhooks/iam/v1alpha1/userdeactivation_webhook.go

```diff
@@ -0,0 +1,126 @@
+		return nil
+	}
+
+	req, err := admission.RequestFromContext(ctx)
+	if err != nil {
+		userdeactivationlog.Error(err, "failed to get admission request from context", "name", ud.GetName())
+		return fmt.Errorf("failed to get request from context: %w", err)
+	}
+
+	// Populate the field with the username present in the access token / UserInfo.
+	ud.Spec.DeactivatedBy = req.UserInfo.Username
+
+	userdeactivationlog.Info("Defaulting deactivatedBy complete", "name", ud.GetName(), "deactivatedBy", ud.Spec.DeactivatedBy)
+
+	return nil
+}
+
+type UserDeactivationValidator struct {
+	client          client.Client
+	systemNamespace string
+}
+
+func (v *UserDeactivationValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	userDeactivation := obj.(*iamv1alpha1.UserDeactivation)
+	userdeactivationlog.Info("Validating UserDeactivation", "name", userDeactivation.Name)
+
+	// Validate that the referenced user exists
+	userName := userDeactivation.Spec.UserRef.Name
+	if userName == "" {
+		return nil, fmt.Errorf("userRef.name is required")
+	}
+
+	// Ensure spec.deactivatedBy matches the requesting user (it should have been defaulted by the mutator)
+	req, reqErr := admission.RequestFromContext(ctx)
+	if reqErr != nil {
+		return nil, fmt.Errorf("failed to get request from context: %w", reqErr)
+	}
+	if userDeactivation.Spec.DeactivatedBy == "" {
+		return nil, fmt.Errorf("spec.deactivatedBy must be set by the system")
+	}
+	if userDeactivation.Spec.DeactivatedBy != req.UserInfo.Username {
+		return nil, fmt.Errorf("spec.deactivatedBy is managed by the system and cannot be set by the client; if provided, it must match the authenticated user")
+	}
+
+	user := &iamv1alpha1.User{}
+	err := v.client.Get(ctx, client.ObjectKey{Name: userName}, user)
+	if errors.IsNotFound(err) {
+		userdeactivationlog.Error(err, "referenced user does not exist", "userName", userName)
+		return nil, fmt.Errorf("referenced user '%s' does not exist", userName)
+	} else if err != nil {
+		userdeactivationlog.Error(err, "failed to validate user reference", "userName", userName)
+		return nil, fmt.Errorf("failed to validate user reference '%s': %w", userName, err)
+	}
+
+	// Ensure there is no existing UserDeactivation for the same user
+	var existingUDList iamv1alpha1.UserDeactivationList
+	if err := v.client.List(ctx, &existingUDList); err != nil {
+		userdeactivationlog.Error(err, "failed to list existing UserDeactivations", "userName", userName)
+		return nil, fmt.Errorf("failed to list existing UserDeactivations for user '%s': %w", userName, err)
+	}
+	for _, existing := range existingUDList.Items {
+		if existing.Spec.UserRef.Name == userName && existing.DeletionTimestamp == nil {
+			userdeactivationlog.Error(fmt.Errorf("a UserDeactivation already exists for user '%s'", userName), "existing UserDeactivation", "name", existing.Name)
+			return nil, fmt.Errorf("a UserDeactivation already exists for user '%s'", userName)
+		}
+	}
```

Comment:
This should result in [an Already Exists error](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#NewAlreadyExists).